nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/test/groovy/org/apache/nifi/web/security/oidc/OidcServiceGroovyTest.groovy

```diff
@@ -0,0 +1,212 @@
+        }
+        mockNFP as NiFiProperties
+    }
+
+    private static JwtService buildJwtService() {
+        def mockJS = new JwtService([:] as KeyService) {
+            @Override
+            String generateSignedToken(LoginAuthenticationToken lat) {
+                signNiFiToken(lat)
+            }
+        }
+        mockJS
+    }
+
+    private static String signNiFiToken(LoginAuthenticationToken lat) {
+        String identity = "mockUser"
+        String USERNAME_CLAIM = "username"
+        String KEY_ID_CLAIM = "keyId"
+        Calendar expiration = Calendar.getInstance()
+        expiration.setTimeInMillis(System.currentTimeMillis() + 10_000)
+        String username = lat.getName()
+
+        return Jwts.builder().setSubject(identity)
+                .setIssuer(lat.getIssuer())
+                .setAudience(lat.getIssuer())
+                .claim(USERNAME_CLAIM, username)
+                .claim(KEY_ID_CLAIM, SIGNING_KEY.getId())
+                .setExpiration(expiration.getTime())
+                .setIssuedAt(Calendar.getInstance().getTime())
+                .signWith(SignatureAlgorithm.HS256, SIGNING_KEY.key.getBytes("UTF-8")).compact()
+    }
+
+    @Test
+    void testShouldStoreJwt() {
+        // Arrange
+        StandardOidcIdentityProvider soip = buildIdentityProviderWithMockInitializedProvider([:])
+
+        OidcService service = new OidcService(soip)
+
+        // Expected JWT
+        logger.info("EXPECTED_JWT: ${MOCK_JWT}")
+
+        // Act
+        service.storeJwt(MOCK_REQUEST_IDENTIFIER, MOCK_JWT)
+
+        // Assert
+        final String cachedJwt = service.getJwt(MOCK_REQUEST_IDENTIFIER)
+        logger.info("Cached JWT: ${cachedJwt}")
+
+        assert cachedJwt == MOCK_JWT
+    }
+
+    @Test
+    void testShouldGetJwt() {
+        // Arrange
+        StandardOidcIdentityProvider soip = buildIdentityProviderWithMockInitializedProvider([:])
+
+        OidcService service = new OidcService(soip)
+
+        // Expected JWT
+        logger.info("EXPECTED_JWT: ${MOCK_JWT}")
+
+        // store the jwt
+        service.storeJwt(MOCK_REQUEST_IDENTIFIER, MOCK_JWT)
+
+        // Act
+        final String retrievedJwt = service.getJwt(MOCK_REQUEST_IDENTIFIER)
+        logger.info("Retrieved JWT: ${retrievedJwt}")
+
+        // Assert
+        assert retrievedJwt == MOCK_JWT
+    }
+
+    @Test
+    void testGetJwtShouldReturnNullWithExpiredDuration() {
+        // Arrange
+        StandardOidcIdentityProvider soip = buildIdentityProviderWithMockInitializedProvider([:])
+
+        final int DURATION = 1
+        final TimeUnit EXPIRATION_UNITS = TimeUnit.SECONDS
+        OidcService service = new OidcService(soip, DURATION, EXPIRATION_UNITS)
+
+        // Expected JWT
+        logger.info("EXPECTED_JWT: ${MOCK_JWT}")
+
+        // Store the jwt
+        service.storeJwt(MOCK_REQUEST_IDENTIFIER, MOCK_JWT)
+
+        // Put thread to sleep
+        long millis = 3 * 1000
+        Thread.sleep(millis)
```

Comment:
With the large number of modules in NiFi, introducing additional delays contributes to longer build times.  Rather than testing in seconds, recommend using setting the expiration in milliseconds and reducing the sleep duration.

Reply:
Thanks for the suggestion, @exceptionfactory . Fixed.